src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +34,10 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(p -> p % 2 == 1 ? numbers.get(p) - 1 : numbers.get(p))
+                .filter(p -> p % 2 == 1).average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .filter(p -> p % 2 == 1)
                .average()
                .orElseThrow(NoSuchElementException::new);
```
![image](https://user-images.githubusercontent.com/20326858/211542678-f35735f0-72cd-492d-ae0c-e9765aa2a856.png)